### DISCUSSION TITLE
How to partially apply a stash?

### DISCUSSION THREAD

**Discussion by gthau** [jesseduffield/lazygit#2203] [Q&A] [answered]
Use case:
I stash several files (e.g. config/env files) and later I want to re-apply the changes of some of the files of a stash entry.

I go to the stash panel, I browse to the stash entry:
- if I press `space` I can apply the whole stash entry.
- if I press `enter` I see the list of files in that stash entry.
    - Then by pressing `space` I can add specific files to the patch  (or even specific lines)
    - I see in the right-most panel that the "custom patch" is building up
    - But then how do I apply this patch?

**Comment by mark2185** [comment, marked as answer]
Go to the `Commits` panel (`4`) or `Local Branches` panel (`3`) and press `ctrl+p`.

Under the menu `x` its description is: `view custom patch options...`

**Reply by gthau** [reply]
Indeed it's `ctrl+p` (`view custom patch options`) and it's accessible in the stash panel when entering a stash (panel with title `Diff files(stash{...})`.

Thank you!

**Reply by mark2185** [reply]
Great! Feel free to mark this as solved! :)
